Userland/Applications/FileManager/PropertiesWindow.cpp

```diff
@@ -251,6 +257,98 @@ ErrorOr<void> PropertiesWindow::create_audio_tab(GUI::TabWidget& tab_widget, Non
+        auto loops = image_decoder->loop_count();
+        auto frames = image_decoder->frame_count();
+        StringBuilder builder;
+        if (loops == 0) {
+            TRY(builder.try_append("Loop indefinitely"sv));
+        } else if (loops == 1) {
+            TRY(builder.try_append("Once"sv));
+        } else {
+            TRY(builder.try_appendff("Loop {} times"sv, loops));
+        }
+        TRY(builder.try_appendff(" ({} frames)"sv, frames));
+
+        animation_text = TRY(builder.to_string());
+    } else {
+        animation_text = "None"_short_string;
+    }
+    tab->find_descendant_of_type_named<GUI::Label>("image_animation")->set_text(move(animation_text));
+
+    auto hide_icc_group = [&tab](String profile_text) {
+        tab->find_descendant_of_type_named<GUI::Label>("image_has_icc_profile")->set_text(profile_text);
+        tab->find_descendant_of_type_named<GUI::Widget>("image_icc_group")->set_visible(false);
+    };
+
+    if (embedded_icc_bytes.has_value()) {
+        auto icc_profile_or_error = Gfx::ICC::Profile::try_load_from_externally_owned_memory(embedded_icc_bytes.value());
+        if (icc_profile_or_error.is_error()) {
+            hide_icc_group(TRY("Present but invalid"_string));
+        } else {
+            auto icc_profile = icc_profile_or_error.release_value();
+            auto extract_string = [](Gfx::ICC::Profile& profile, Gfx::ICC::TagSignature signature) -> ErrorOr<String> {
+                auto maybe_tag_data = profile.tag_data(signature);
+                if (!maybe_tag_data.has_value())
+                    return String {};
+                auto& tag_data = maybe_tag_data.release_value();
+                if (tag_data.type() == Gfx::ICC::MultiLocalizedUnicodeTagData::Type) {
+                    auto& multi_localized_unicode = static_cast<Gfx::ICC::MultiLocalizedUnicodeTagData const&>(tag_data);
+                    // Try to find 'en-US', otherwise any 'en' language, otherwise the first record.
+                    Optional<String> en_string;
+                    constexpr u16 language_en = ('e' << 8) + 'n';
+                    constexpr u16 country_us = ('U' << 8) + 'S';
+                    for (auto const& record : multi_localized_unicode.records()) {
+                        if (record.iso_639_1_language_code == language_en) {
+                            if (record.iso_3166_1_country_code == country_us)
+                                return record.text;
+                            en_string = record.text;
+                        }
+                    }
+                    return en_string.value_or(multi_localized_unicode.records().first().text);
+                } else if (tag_data.type() == Gfx::ICC::TextDescriptionTagData::Type) {
+                    auto& text_description = static_cast<Gfx::ICC::TextDescriptionTagData const&>(tag_data);
+                    // FIXME: We should probably pay attention to the value of unicode_language_code() ?
```

Comment:
Eh, not sure what we'd do with it.

The spec says (https://color.org/specification/ICC.1-2001-04.pdf, "6.5.17 textDescriptionType"):

"The 7-bit ASCII description is to be an invariant, nonlocalizable name for consistent reference. It is preferred that both the Unicode and ScriptCode structures be properly localized."

Since serenity isn't localized, maybe we should prefer `ascii_description()` anyways?